privval/sc_file.go

```diff
@@ -180,8 +187,9 @@ func (pv *SCFilePV) OnStop() {
 	pv.HTTP.Close()
 
 	// Save rank to last_rank.json file if the shutdown was not self-induced.
-	if err := pv.Save(config.Dir(), pv.Logger); err != nil {
-		fmt.Printf("[ERR] signctrl: couldn't save rank to %v: %v", LastRankFile, err)
+	pv.State.LastRank = pv.GetRank()
+	if err := pv.State.Save(config.Dir()); err != nil {
+		fmt.Printf("[ERR] signctrl: couldn't save state to %v: %v", config.StateFile, err)
 		os.Exit(1)
```

Comment:
should we really exit here? Can't we pass the error up? I don't thin it's a good idea to exit at many places in the code, will make writing tests hard.

Reply:
Agreed. I added an error return value to the OnStop() method of the SignCtrled interface, that should do it.